registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/main/java/io/mosip/registration/processor/stages/packetclassifier/tagging/impl/IDObjectFieldsTagGenerator.java

```diff
@@ -0,0 +1,141 @@
+package io.mosip.registration.processor.stages.packetclassifier.tagging.impl;
+
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.registration.processor.core.constant.MappingJsonConstants;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+import io.mosip.registration.processor.core.util.JsonUtil;
+import io.mosip.registration.processor.packet.storage.exception.ParsingException;
+import io.mosip.registration.processor.packet.storage.utils.Utilities;
+import io.mosip.registration.processor.stages.packetclassifier.dto.FieldDTO;
+import io.mosip.registration.processor.stages.packetclassifier.tagging.TagGenerator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnExpression(value = "'${mosip.regproc.packet.classifier.tag-generators}'.contains('MosipIDObjectFields')")
+public class IDObjectFieldsTagGenerator implements TagGenerator {
+
+    /**
+     * These field names should be as in keys of registraion-processor-identity.json file Identity segment
+     * and should have proper default source configured
+     */
+    @Value("#{'${mosip.regproc.packet.classifier.tagging.idobjectfields.mapping-field-names}'.split(',')}")
+    private List<String> mappingFieldNames;
+
+    /** The tag name that will be prefixed with every idobjectfield tags */
+    @Value("${mosip.regproc.packet.classifier.tagging.idobjectfields.tag-name-prefix:ID_OBJECT-}")
+    private String tagNamePrefix;
+
+    /** The language that should be used when dealing with field type that has values in multiple languages */
+    @Value("${mosip.regproc.packet.classifier.tagging.idobjectfields.tag-language}")
```

Comment:
without a default value this becomes mandatory. Do we need this?

Reply:
yes, since in simpletype is a array of label and value, we need this to get the correct value

Reply:
Will use the primary lang

Reply:
changes done